tests/flytekit/unit/core/test_dataclass.py

```diff
@@ -41,3 +62,816 @@ class MyDC(DataClassJSONMixin):
+        flyte_types=FlyteTypes(
+            flytefile=ff,
+            flytedir=fd,
+            structured_dataset=sd,
+        ),
+        list_flyte_types=[FlyteTypes(
+            flytefile=ff,
+            flytedir=fd,
+            structured_dataset=sd,
+        )],
+        dict_flyte_types={
+            "a": FlyteTypes(
+                flytefile=ff,
+                flytedir=fd,
+                structured_dataset=sd,
+            ),
+            "b": FlyteTypes(
+                flytefile=ff,
+                flytedir=fd,
+                structured_dataset=sd)},
+        optional_flyte_types=FlyteTypes(
+            flytefile=ff,
+            flytedir=fd,
+            structured_dataset=sd,
+        ),
+    )
+
+    lt = TypeEngine.to_literal_type(NestedFlyteTypes)
+    lv = TypeEngine.to_literal(ctx, o, NestedFlyteTypes, lt)
+    assert lv is not None
+    pv = TypeEngine.to_python_value(ctx, lv, NestedFlyteTypes)
+    assert isinstance(pv, NestedFlyteTypes)
+    DataclassTransformer().assert_type(NestedFlyteTypes, pv)
+
+# For mashumaro dataclasses mixins, it's equal to use @dataclasses only
+def test_mashumaro_dataclasses_json_mixin_with_python_types():
+    @dataclass
+    class DC(DataClassJSONMixin):
+        string: Optional[str] = None
+
+    @dataclass
+    class DCWithOptional(DataClassJSONMixin):
+        string: Optional[str] = None
+        dc: Optional[DC] = None
+        list_dc: Optional[List[DC]] = None
+        list_list_dc: Optional[List[List[DC]]] = None
+        dict_dc: Optional[Dict[str, DC]] = None
+        dict_dict_dc: Optional[Dict[str, Dict[str, DC]]] = None
+        dict_list_dc: Optional[Dict[str, List[DC]]] = None
+        list_dict_dc: Optional[List[Dict[str, DC]]] = None
+
+    @task
+    def t1() -> DCWithOptional:
+        return DCWithOptional(string="a", dc=DC(string="b"),
+                              list_dc=[DC(string="c"), DC(string="d")],
+                              list_list_dc=[[DC(string="e"), DC(string="f")]],
+                              list_dict_dc=[{"g": DC(string="h"), "i": DC(string="j")},
+                                            {"k": DC(string="l"), "m": DC(string="n")}],
+                              dict_dc={"o": DC(string="p"), "q": DC(string="r")},
+                              dict_dict_dc={"s": {"t": DC(string="u"), "v": DC(string="w")}},
+                              dict_list_dc={"x": [DC(string="y"), DC(string="z")],
+                                            "aa": [DC(string="bb"), DC(string="cc")]},)
+
+    @task
+    def t2() -> DCWithOptional:
+        return DCWithOptional()
+
+    dc1 = t1()
+    dc2 = t2()
+    assert dc1.string == "a"
+    assert dc1.dc.string == "b"
+    assert dc1.list_dc[0].string == "c"
+    assert dc1.list_dc[1].string == "d"
+    assert dc1.list_list_dc[0][0].string == "e"
+    assert dc1.list_list_dc[0][1].string == "f"
+    assert dc1.list_dict_dc[0]["g"].string == "h"
+    assert dc1.list_dict_dc[0]["i"].string == "j"
```

Comment:
Could we create a variable for DCWithOptional?

```python
output = DCWithOptional(string="a", dc=DC(string="b"),
                              list_dc=[DC(string="c"), DC(string="d")],
                              list_list_dc=[[DC(string="e"), DC(string="f")]],
                              list_dict_dc=[{"g": DC(string="h"), "i": DC(string="j")},
                                            {"k": DC(string="l"), "m": DC(string="n")}],
                              dict_dc={"o": DC(string="p"), "q": DC(string="r")},
                              dict_dict_dc={"s": {"t": DC(string="u"), "v": DC(string="w")}},
                              dict_list_dc={"x": [DC(string="y"), DC(string="z")],
                                            "aa": [DC(string="bb"), DC(string="cc")]},)
```

And then compare the output with t1()

```python
assert output == t1()
```

Reply:
Yes, thank you